src/pudl/transform/ferc1.py

```diff
@@ -1296,80 +1360,69 @@ def process_xbrl_metadata(self: Self, xbrl_metadata_json) -> pd.DataFrame:
-            self.deduplicate_xbrl_factoid_xbrl_metadata(tbl_meta)
-            .pipe(self.apply_xbrl_calculation_fixes)
-            .pipe(self.remove_duplicated_calculation_components)
+            .pipe(self.deduplicate_xbrl_factoid_xbrl_metadata)
         )
+        return tbl_meta
 
-        # Add intra-table calculation flag after all calculation components adjusted.
-        tbl_meta["intra_table_calc_flag"] = tbl_meta.calculations.apply(
-            check_for_other_source_tables, native_table=self.table_id.value
-        ).astype(pd.BooleanDtype())
+    @cache_df(key="process_xbrl_metadata")
+    def process_xbrl_metadata(
+        self: Self, xbrl_meta_raw: pd.DataFrame, xbrl_calculations: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Process XBRL metadata after the calculations have been cleaned.
 
-        tbl_meta = tbl_meta.pipe(self.add_calculation_correction_components)
+        Add ``row_type_xbrl`` column and create ``xbrl_factoid`` records for the
+        calculation corrections.
 
+        Args:
+            xbrl_meta_raw: Dataframe of relatively unprocessed metadata. Result of
+                :meth:`pre_process_xbrl_metadata`
+            xbrl_calculations: Dataframe of calculation components. Result of
+                :meth:`process_xbrl_metadata_calculations`
+        """
+        tbl_meta = xbrl_meta_raw.drop(columns=["calculations"]).set_index(
+            ["xbrl_factoid"]
+        )
         # Flag metadata record types
-        tbl_meta = tbl_meta.assign(
-            row_type_xbrl=lambda x: np.where(
-                x.calculations != "[]", "calculated_value", "reported_value"
+        tbl_meta.loc[:, "row_type_xbrl"] = (
+            (
+                xbrl_calculations.groupby(["xbrl_factoid_parent"])[
+                    ["table_name", "xbrl_factoid"]
+                ].count()
+                == 0
             )
-        ).astype({"row_type_xbrl": pd.StringDtype(), "calculations": pd.StringDtype()})
-
-        # Create metadata records for the calculation correction factoids
-        correction_meta = (
-            tbl_meta[tbl_meta.row_type_xbrl == "calculated_value"]
-            .copy()
-            .assign(
-                calculations="[]",
-                intra_table_calc_flag=True,
-                row_type_xbrl="correction",
-                xbrl_factoid=lambda x: x.xbrl_factoid + "_correction",
+            .all(axis="columns")
+            .replace({True: "reported_value", False: "calculated_value"})
+            .astype(pd.StringDtype())
+        )
+        tbl_meta.loc[:, "intra_table_calc_flag"] = (
+            # make a temp bool col to check if all the componets are intra table
+            # should the non-calc guys get a null or a true here? rn its true bc fillna
+            xbrl_calculations.assign(
+                intra_table_calc_comp_flag=lambda x: (
+                    self.table_id.value == x.table_name.fillna(self.table_id.value)
+                )
             )
```

Comment:
rn this column is being generated in two places! eek. we need it in the calc components table in two places but its really a factoid-level piece of info?

Reply:
i'm keeping this column in both tables. instead of fully generating it in two places, now i'm effectively reading it in from the calc table here in the metadata.